src/internal/transactionenv/env.go

```diff
@@ -191,24 +150,18 @@ type PpsTransactionServer interface {
 // without leaving the context of a transaction.  This is a separate object
 // because there are cyclic dependencies between APIServer instances.
 type TransactionEnv struct {
-	serviceEnv serviceenv.ServiceEnv
-	txnServer  TransactionServer
-	authServer AuthTransactionServer
-	pfsServer  PfsTransactionServer
-	ppsServer  PpsTransactionServer
+	txnServer TransactionServer
+	pfsServer PfsTransactionServer
+	ppsServer PpsTransactionServer
 }
 
 // Initialize stores the references to APIServer instances in the TransactionEnv
 func (env *TransactionEnv) Initialize(
-	serviceEnv serviceenv.ServiceEnv,
```

Comment:
Removing the serviceenv here is necessary to resolve a circular dependency, instead we pass in the etcd client (soon to be postgres client) for each new transaction being created.